crates/sui-graphql-rpc/src/server/watermark_task.rs

```diff
@@ -60,13 +66,22 @@ impl WatermarkTask {
     }
 
     pub(crate) async fn run(&self) {
+        let mut interval = tokio::time::interval(self.sleep);
+        // We start the task by first finding & setting the chain identifier
+        // so that it can be used in all requests.
+        let cached = self.get_and_cache_chain_identifier(&mut interval).await;
+        // In case we received a cancel signal, we don't want to start the classic watermark task.
+        if !cached {
+            return;
+        };
```

Comment:
Is it necessary to have this `cached` check? Presumably it's using the same cancellation token as is checked below, so we could rely on the test within the `tokio::select!` right?

```suggestion
        self.get_and_cache_chain_identifier(&mut interval).await;
```

Reply:
Yup you're right, I did that before I understood the full context on how it works.